### PR TITLE
Implement `ShouldCreate` function to to component extensions

### PR DESCRIPTION
**Pull request by finobinos**

### CODE REVIEW

File: modules/component/init.lua [hunk 1]
```diff
@@ -115,10 +115,14 @@ local function InvokeExtensionFn(component, extensionList, fnName: string)
 	for _,extension in ipairs(extensionList) do
 		local fn = extension[fnName]
 		if type(fn) == "function" then
-			fn(component)
+		 return fn(component)
```
Comment (EncodedVenom) [inline modules/component/init.lua:118]: This code doesn't allow multiple extensions with the same function to run. I feel like this should be returning an array instead.

**Comment by EncodedVenom** [comment]
This pull does not add anything to the existing Extension type. May want to look into that.

**Comment by finobinos** [comment]
> This pull does not add anything to the existing Extension type. May want to look into that.

Extensions would still be slightly limited, but a `ShouldCreate` function should def improve it. @EncodedVenom

File: modules/component/init.lua (hunk 1 above)
Comment (finobinos) [reply modules/component/init.lua]: Thanks! This has been solved, see: https://github.com/Sleitnick/RbxUtil/pull/15/commits/7ac686c916e11256d63967f3b5b955f87392fee4

**Comment by Sleitnick** [comment]
Thanks for the initial PR here! I've made another which accomplishes this but with some slight modifications
https://github.com/Sleitnick/RbxUtil/pull/17

**Pull request closed without merging**